src/category_theory/groupoid/subgroupoid.lean

```diff
@@ -260,13 +336,39 @@ variable (X : ∀ c d : C, set (c ⟶ d))
 def generated : subgroupoid C :=
 Inf {S : subgroupoid C | ∀ c d, X c d ⊆ S.arrows c d}
 
+lemma generated_contains : ∀ c d : C, X c d ⊆ (generated X).arrows c d :=
+begin
+  dsimp only [generated, Inf],
+  simp only [subset_Inter₂_iff],
+  exact λ c d S hS, hS c d,
+end
+
 /-- The normal sugroupoid generated by the set of arrows `X` -/
 def generated_normal : subgroupoid C :=
 Inf {S : subgroupoid C | (∀ c d, X c d ⊆ S.arrows c d) ∧ S.is_normal}
 
+lemma generated_normal_contains_generated : generated X ≤ generated_normal X :=
+begin
+  apply @Inf_le_Inf (subgroupoid C) _,
+  exact λ S ⟨h,_⟩, h,
+end
+
 lemma generated_normal_is_normal : (generated_normal X).is_normal :=
 Inf_is_normal _ (λ S h, h.right)
 
+lemma is_normal.containing_iff_generated_normal_le {S : subgroupoid C} (Sn : S.is_normal) :
+  (∀ c d, X c d ⊆ S.arrows c d) ↔ (generated_normal X) ≤ S :=
+begin
+  split,
+  { rintro h,
+    apply @Inf_le (subgroupoid C) _,
+    exact ⟨h,Sn⟩, },
+  { rintro h c d,
+    let h' := generated_normal_contains_generated X,
+    rw le_iff at h h',
+    exact ((generated_contains X c d).trans (@h' c d)).trans (@h c d), },
+end
+
```

Comment:
Just noting that the `generated` and `generated_normal` also forms a `galois_insertion` with the forgetful map `subgroupoid → subquiver`. It would be useful for giving the lattice structure on subgroupoids / normal subgroupoids and other stuff. You might want to take a look (not necessarily in this PR though).

Reply:
I wanted to implement a structure for subquivers at some point: this will provide a good reason to actually do it.